spec/tbd_tests_spec.rb

```diff
@@ -1185,119 +1224,137 @@
-    # sun exposure AFTER setting boundary conditions to "Foundation", e.g.
-    #
-    #   expect(oa15.setOutsideBoundaryCondition("Foundation")).to be true
-    #   expect(oa15.setWindExposure("NoWind")).to be true
-    #   expect(oa15.setSunExposure("NoSun")).to be true
-    #
-    # Alas, both "exposures" end up being reset in the saved OSM. One solution
-    # is to first set the 'wall height above grade' value to 0. Works.
-    kf = model.getFoundationKivas.first
-    expect(kf.isWallHeightAboveGradeDefaulted).to be true
-    expect(kf.wallHeightAboveGrade.round(1)).to eq(0.2)
-    expect(kf.setWallHeightAboveGrade(0)).to be true
-    expect(kf.isWallHeightAboveGradeDefaulted).to be false
-    expect(kf.wallHeightAboveGrade.round).to eq(0)
-    expect(oa15.setWindExposure("NoWind")).to be true
-    expect(oa15.setSunExposure("NoSun")).to be true
-
-    found_floor = false
-    found_wall  = false
+      construction = oaWALL.construction.get
+      expect(oaWALL.outsideBoundaryCondition.downcase).to eq("outdoors")
+      expect(oaWALL.setOutsideBoundaryCondition("Foundation")).to be true
+      expect(oaWALL.setConstruction(construction)).to be true
 
-    surfaces.each do |id, surface|
-      next unless surface.key?(:kiva)
+      srfIDs << oaID
 
-      expect(id).to eq("Open area 1 Floor").or eq("Openarea 1 Wall 5")
+      argh              = {}
+      argh[:option    ] = "(non thermal bridging)"
+      argh[:gen_kiva  ] = true
+      argh[:reset_kiva] = true
 
-      if id == "Open area 1 Floor"
-        expect(surface[:kiva]).to eq(:basement)
-        expect(surface).to have_key(:exposed)
-        expect(surface[:exposed]).to be_within(TOL).of(12.59) # not 8.70
-        found_floor = true
-      else
-        expect(surface[:kiva]).to eq("Open area 1 Floor")
-        found_wall = true
+      json     = TBD.process(model, argh)
+      expect(json).to be_a(Hash)
+      expect(json).to have_key(:io)
+      expect(json).to have_key(:surfaces)
+      io       = json[:io      ]
+      surfaces = json[:surfaces]
+      expect(TBD.info?).to be true
+      expect(TBD.logs.size).to eq(i + 1)
+
+      TBD.logs.each do |lg|
+        expect(lg[:message]).to include("Purged KIVA objects from ")
       end
-    end
 
-    expect(found_floor).to be true
-    expect(found_wall).to be true
+      expect(surfaces).to be_a(Hash)
+      expect(surfaces.size).to eq(56)
+      expect(io).to be_a(Hash)
+      expect(io).to have_key(:edges)
+      expect(io[:edges].size).to eq(105 - 2 * i)
+      expect(model.foundationKivaSettings).to_not be_empty
+      expect(model.getSurfacePropertyExposedFoundationPerimeters.size).to eq(1)
+      expect(model.getFoundationKivas.size).to eq(1) # !4 ... previously purged
+
+      perimeter = model.getSurfacePropertyExposedFoundationPerimeters.first
+      expect(perimeter.totalExposedPerimeter).to_not be_empty
+      expect(perimeter.totalExposedPerimeter.get.round(2)).to eq(exp.round(2))
```

Comment:
Regardless of the number of _outdoor-facing_ vs _foundation-facing_ walls, exposed wall perimeter remains constant.